packages/realm-web/src/Credentials.ts

```diff
@@ -53,10 +53,11 @@ type SimpleObject = Record<string, unknown>;
 export class Credentials<PayloadType extends SimpleObject = SimpleObject> implements Realm.Credentials<PayloadType> {
   /**
    * Creates credentials that logs in using the [Anonymous Provider](https://docs.mongodb.com/realm/authentication/anonymous/).
+   * @param reuse - Reuse any existing anonymous user already logged in.
    * @returns The credentials instance, which can be passed to `app.logIn`.
    */
-  static anonymous(): Credentials<AnonymousPayload> {
-    return new Credentials<AnonymousPayload>("anon-user", "anon-user", {});
+  static anonymous(reuse = true): Credentials<AnonymousPayload> {
```

Comment:
Current version is 2.0.0, and bumping to 3.0.0 might be drastic. If we say `reuse = false` and say we will switch to `reuse = true` in the future, we can avoid the bump.

The next big project for Realm Web will be enhanced MongoDB data access project, and it is - for me - unclear if it will come will any breaking changes.

Reply:
We could also brand this as a bug fix as `reuse = true` was the intended behavior from the start and creating many users is most likely not what developers intend to do.

Reply:
I'd be okay with changing the default to `false`, but that would deviate from the "realm" package, which could potentially be confusing 🤔
https://github.com/realm/realm-js/blob/0bd8ee486780431594724e3ff4988aa0853e7a4e/packages/realm/src/app-services/Credentials.ts#L56

But I agree, we could bundle the change to `true` into a larger breaking change (possibly as part of the project to extract the App Services client).

Reply:
The docs from the `realm` package is a bit self-contradictory on this subject: https://www.mongodb.com/docs/realm-sdks/js/latest/classes/export_.Credentials.html#anonymous

The documentation on the function itself:
> Creates credentials for an anonymous user. These can only be used once - using them a second time will result in a different user being logged in. If you need to get a user that has already logged in with the Anonymous credentials, use [App.currentUser](https://www.mongodb.com/docs/realm-sdks/js/latest/classes/export_.App-1.html#currentUser) or [App.allUsers](https://www.mongodb.com/docs/realm-sdks/js/latest/classes/export_.App-1.html#allUsers).

The documentation of the optional argument:

> `reuse: boolean = true`
> Reuse any existing anonymous user already logged in.

Reply:
I moved it to "fixed".